projects/media-viewer/src/lib/directives/drag-scroll.directive.ts

```diff
@@ -0,0 +1,49 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { Position } from './model/position.model';
+import { Scroll } from './model/scroll.model';
+
+@Directive({
+  selector: '[dragScroll]'
+})
+export class DragScrollDirective {
+
+  originalScroll: Scroll;
+  private pointerDown = false;
+
+  @Input() enableScroll = true;
+  @Input() scrollAxis = '';
```

Comment:
I would make this explicitly, scrollX, and scrollY, and if it's set it means it's enabled